src/main/kotlin/lotto/domain/WinningLotto.kt

```diff
@@ -1,6 +1,20 @@
 package lotto.domain
 
-class WinningLotto(private val winningNumbers: Lotto) {
-    fun getMatchesCount(lotto: Lotto): Int = lotto.numbers.count { num -> winningNumbers.numbers.contains(num) }
-    fun getMatchesCount(lottos: List<Lotto>): List<Int> = lottos.map { lotto -> getMatchesCount(lotto) }
+import lotto.dto.MatchResultDto
+
+class WinningLotto(private val winningNumbers: Lotto, private val bonusNumber: LottoNumber) {
+
+    init {
+        require(!winningNumbers.numbers.contains(bonusNumber)) { "보너스 볼은 당첨 번호랑 같을 수 없습니다." }
+    }
+
+    fun getMatchResult(lotto: Lotto): MatchResultDto =
+        MatchResultDto(getCountOfMatch(lotto), matchBonus(lotto))
+
+    fun getMatchResult(lottos: List<Lotto>): List<MatchResultDto> =
+        lottos.map { getMatchResult(it) }
+
+    private fun getCountOfMatch(lotto: Lotto): Int = lotto.numbers.count { winningNumbers.numbers.contains(it) }
```

Comment:
```suggestion
    private fun getCountOfMatch(lotto: Lotto): Int = lotto.matchingCount(winningNumbers)
```
train wreck한 코드는 유지보수에 좋지 않다고 생각해요. 변경 해볼까요?